crates/holochain_deterministic_integrity/src/hdi.rs

```diff
@@ -150,9 +148,12 @@ impl HdiT for HostHdi {
     fn zome_info(&self, _: ()) -> ExternResult<ZomeInfo> {
         host_call::<(), ZomeInfo>(__zome_info, ())
     }
-    #[cfg(feature = "trace")]
     fn trace(&self, trace_msg: TraceMsg) -> ExternResult<()> {
-        host_call::<TraceMsg, ()>(__trace, trace_msg)
+        if cfg!(feature = "trace") {
+            host_call::<TraceMsg, ()>(__trace, trace_msg)
+        } else {
+            Ok(())
```

Comment:
Maybe this should be an error instead of silently failing?

Reply:
True